docs/lang/articles/advanced/layout.md

```diff
@@ -2,30 +2,59 @@
 sidebar_position: 2
 ---
 
-# Advanced dense layouts
+# Fields (advanced)
 
-Fields can be _placed_
-in a specific shape and _layout_. Defining a proper layout can be
-critical to performance, especially for memory-bound applications. A
-carefully designed data layout can significantly improve cache/TLB-hit
-rates and cacheline utilization. Although when performance is not the
-first priority, you probably don't have to worry about it.
+This section introduces some advanced features of Taichi fields.
+Make sure you have gone through [Fields](../basic/field).
 
-Taichi decouples algorithms from data layouts, and the Taichi compiler
-automatically optimizes data accesses on a specific data layout. These
-Taichi features allow programmers to quickly experiment with different
-data layouts and figure out the most efficient one on a specific task
-and computer architecture.
+## Packed mode
 
-In Taichi, the layout is defined in a recursive manner.
-We suggest starting with the default layout specification (simply
-by specifying `shape` when creating fields using
-`ti.field/ti.Vector.field/ti.Matrix.field`), and then migrate to more
-advanced layouts using the `ti.root.X` syntax if necessary.
+By default, all non-power-of-two dimensions of a field are automatically
+padded to a power of two. For instance, a field of shape `(18, 65)` will
+have internal shape `(32, 128)`. Although the padding has many benefits
+such as allowing fast and convenient bitwise operations for coordinate
+handling, it will consume potentially much more memory than you originally
+think.
 
-## From `shape` to `ti.root.X`
+If you indeed want smaller memory usage, you could use the optional packed
+mode. In packed mode, no more padding will be applied so a field will not
+have a larger internal shape when some of its dimensions are not power-of-two.
+The downside is that the runtime performance will regress slightly.
```

Comment:
```suggestion
If you would like to reduce memory usage, you can use the optional packed
mode. In packed mode, no padding is applied such that a field does not
have a larger internal shape than when some of its dimensions are not powers oftwo.
The downside is that the runtime performance will regress slightly.
```

Reply:
Here I meant `a field does not have a larger internal shape than the defined shape when some of its dimensions are not powers of two`. Is there a better way to express that?

Reply:
I think this is already clear enough 😄